Cannot build from master

````
> cargo build
   Compiling winapi v0.3.4
...
   Compiling rustc-ap-rustc_errors v121.0.0
error[E0308]: mismatched types
   --> src/bin/main.rs:361:20
    |
361 |         if kind == "default" {
    |                    ^^^^^^^^^ expected enum `std::option::Option`, found reference
    |
    = note: expected type `std::option::Option<std::string::String>`
               found type `&'static str`

error[E0308]: mismatched types
   --> src/bin/main.rs:362:62
    |
362 |             return Ok(Operation::ConfigOutputDefault { path: path.clone() });
    |                                                              ^^^^^^^^^^^^ expected struct `std::string::String`, found reference
    |
    = note: expected type `std::option::Option<std::string::String>`
               found type `std::option::Option<&std::string::String>`

error[E0308]: mismatched types
   --> src/bin/main.rs:363:26
    |
363 |         } else if kind = "minimal" {
    |                          ^^^^^^^^^ expected enum `std::option::Option`, found reference
    |
    = note: expected type `std::option::Option<std::string::String>`
               found type `&'static str`

error[E0308]: mismatched types
   --> src/bin/main.rs:363:19
    |
363 |         } else if kind = "minimal" {
    |                   ^^^^^^^^^^^^^^^^
    |                   |
    |                   expected bool, found ()
    |                   help: try comparing for equality: `kind == "minimal"`
    |
    = note: expected type `bool`
               found type `()`

error[E0308]: mismatched types
   --> src/bin/main.rs:396:9
    |
396 |         minimal_config_path,
    |         ^^^^^^^^^^^^^^^^^^^ expected struct `std::string::String`, found reference
    |
    = note: expected type `std::option::Option<std::string::String>`
               found type `std::option::Option<&std::string::String>`

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0308`.
error: Could not compile `rustfmt-nightly`.
warning: build failed, waiting for other jobs to finish...
error: build failed
````

Compiler info
````
> cargo --version
cargo 1.27.0-nightly (9e53ac6e6 2018-05-07)
> rustc --version
rustc 1.27.0-nightly (acd3871ba 2018-05-10)
````
@nrc, you probably wanna see this.

--- a/src/bin/main.rs
+++ b/src/bin/main.rs
@@ -355,12 +355,11 @@ fn determine_operation(matches: &Matches) -> FmtResult<Operation> {
     }
 
     let mut minimal_config_path = None;
-    if matches.opt_present("print-config") {
-        let kind = matches.opt_str("print-config");
-        let path = matches.free.get(0);
+    if let Some(ref kind) = matches.opt_str("print-config") {
+        let path = matches.free.get(0).cloned();
         if kind == "default" {
-            return Ok(Operation::ConfigOutputDefault { path: path.clone() });
-        } else if kind = "minimal" {
+            return Ok(Operation::ConfigOutputDefault { path });
+        } else if kind == "minimal" {
             minimal_config_path = path;
             if minimal_config_path.is_none() {
                 println!("WARNING: PATH required for `--print-config minimal`");

